app/services/teacher_training_public_api/sync_courses.rb

```diff
@@ -6,7 +6,7 @@ class SyncCourses
     sidekiq_options retry: 3, queue: :low_priority
 
     def perform(provider_id, recruitment_cycle_year, run_in_background: true)
-      @provider = ::Provider.find(provider_id)
+      @provider = ::Provider.find_by!(id: provider_id)
```

Comment:
`find` will throw a `RecordNotFound` exception — no need to change to `find_by!`